### DISCUSSION TITLE
How to access avidemux from the command line?

### DISCUSSION THREAD

**Discussion by cyberdynesoftware** [Homebrew/discussions#631] [Casks]
I've installed the cask for avidemux and it works well when run from Programs. But now I want to do some batch processing with avidemux and noticed that it's not available on the command line:

```
% avidemux
zsh: command not found: avidemux
```

I vaguely remember having this problem before with another cask, I think it was dosbox. When I installed it as a cask, it wasn't available at the command line.

Is this maybe a general problem with casks? Is there a general to solution to it? Can I just create a symbolic link in `/usr/local/bin`? But to where exactly?

**Comment by gromgit** [comment]
In general, Casks use whatever installer program the upstream provider releases, or just puts the provided app bundle in the correct location. There's no general solution to this; some upstreams may not even include CLI binaries in their app bundles.

For the specific case for `avidemux`, there are two CLI binaries in the app bundle, so I've raised a PR to symlink them into the standard Homebrew hierarchy. You may wish to look at that PR to see how it's done, so if you encounter this with any other casks, you can make your own (first?) contribution: https://github.com/Homebrew/homebrew-cask/pull/98865

**Reply by gromgit** [reply]
PR has been merged, so after running the following commands, you should see a couple of additions:
```bash
$ brew update
[...]
==> Updated Casks
avidemux   [...]

$ brew reinstall avidemux
[...]
==> Installing Cask avidemux
==> Moving App 'Avidemux_2.7.6.app' to '/Applications/Avidemux_2.7.6.app'
==> Linking Binary 'avidemux_cli' to '/usr/local/bin/avidemux'
==> Linking Binary 'avidemux_jobs' to '/usr/local/bin/avidemux_jobs'
```

**Reply by cyberdynesoftware** [reply]
Thank you very much, the avidemux executible is now available on the command line. Unfortunately, it looks like that's not enough. avidemux seems to be looking for other stuff it can't find. This is the output from avidemux:

```
[Script] Scanning directory /usr/local/bin/../lib/ADM_plugins6/scriptEngines
[Script] Cannot parse plugin
[Script] Scanning directory /usr/local/bin/../lib/ADM_plugins6/scriptEngines/cli
[Script] Cannot parse plugin
[ADM_ad_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/audioDecoder
[ADM_ad_plugin] Cannot parse plugin
 [ADM_av_loadPlugins] 20:07:35-594  [ADM_av_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/audioDevices
 [ADM_av_loadPlugins] 20:07:35-594  [ADM_av_plugin] Cannot parse plugin
[ADM_ae_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/audioEncoders
[ADM_ae_plugin] Cannot parse plugin
[ADM_dm_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/demuxers
[ADM_av_plugin] Cannot parse plugin
[ADM_mx_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/muxers
[ADM_av_plugin] Cannot parse plugin
[ADM_vd6_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/videoDecoders
[ADM_vd6_plugin] Cannot parse plugin
[ADM_ve6_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/videoEncoders
[ADM_ve6_plugin] Cannot parse plugin
[ADM_ve6_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/videoEncoders/cli
[ADM_ve6_plugin] Cannot parse plugin
[ADM_vf_plugin] Scanning directory /usr/local/bin/../lib/ADM_plugins6/videoFilters
[ADM_vf_plugin] Cannot parse plugin
[ADM_vf_plugin] Cannot parse plugin
```

**Reply by cyberdynesoftware** [reply]
I found a ADM_plugins6 folder in the MacOS folder of avidemux and symlinked that to /usr/local/lib, which did the trick. Now I need to investigate how to add that to the avidemux ruby script.

**Reply by gromgit** [reply]
Hmmm. Since there's [no `library` stanza](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/all_stanzas.md), you'll probably need to write a `postflight` stanza to `File.symlink` it if it doesn't already exist, and an `uninstall_postflight` stanza to `File.delete` it. See the `meshlab` cask for an example of the former.
